strings/base_implements.h

```diff
@@ -1218,6 +1218,35 @@ namespace winrt::impl
     };
 #endif
 
+    template<typename T>
+    class has_initializer
+    {
+        template <typename U, typename = decltype(std::declval<U>().InitializeComponent())> static constexpr bool get_value(int) { return true; }
+        template <typename> static constexpr bool get_value(...) { return false; }
+
+    public:
+        static constexpr bool value = get_value<T>(0);
+    };
+
+
+    template <typename T>
+    auto initialize(T* instance)
```

Comment:
We could fold `new heap_implements<D>` into this function and simplify it:

```cpp
template<typename T, typename... Args>
auto create_and_initialize(Args&&... args)
{
    auto instance = std::make_unique<heap_implements<T>>(std::forward<Args>(args)...);
    if constexpr (has_initializer<T>::value)
    {
        instance->InitializeComponent();
    }

    return instance.release();
}
```

Reply:
That would use `delete` which is incorrect.

Reply:
That was sent before the change to `->Release()` - now we can change it for `com_ptr` to be correct again

Reply:
```cpp
template<typename T, typename... Args>
T* create_and_initialize(Args&&... args)
{
    com_ptr<T> instance;
    instance.attach(new heap_implements<T>(std::forward<Args>(args)...);

    if constexpr (has_initializer<T>::value)
    {
        instance->InitializeComponent();
    }

    return instance.release();
}
```